deps/getdns/getdns.gyp

```diff
@@ -0,0 +1,100 @@
+{
+  'targets': [
+    {
+      'target_name': 'getdns',
+      'type': 'static_library',
+      'include_dirs': [
+        'src',
+        '<(INTERMEDIATE_DIR)',
+        'src/util/auxiliary',
+        'src/openssl',
+        'src/tls',
+        'src/yxml',
+      ],
+      'libraries': ['-lunbound'],
```

Comment:
when i try to build this, the linker fails saying all the libunbound functions are undefined /cc @nodejs/node-gyp

Reply:
What is your question? libunbound is a dependency of getdns, yes.

I don't want to discourage you but I don't see this PR going anywhere. getdns also depends on libidn or libidn2 (non-optionally, AFAICT) and those are GPL.

Reply:
My apologies, I was wrong about libidn being mandatory - it can be disabled by passing `-DUSE_LIBIDN2=OFF` to cmake.

Node.js already does IDNA conversion, you just need to undo the removal of the toASCII / toUnicode calls in this PR. I could also open an upstream PR to teach it about ICU.

As to libunbound: that's a thornier issue. It's part of unbound and pretty big. sloccount says 112,000 lines of C.

I don't think it'd be easy to break out just the parts we want and this PR is already pretty top-heavy at 86,000 lines added.

Reply:
good to know, I left the toASCII in (it's just moved to internal/dns/modern.js).

I guess I can also look into censoring unbound. what a rabbit hole this has turned into :D

Reply:
Is current PR blocked by linker fails ?

Reply:
i need to vendor libunbound as well, and i haven't had the time to figure that out. if only we could use submodules and cmake :(